### PR TITLE
Implement hooked access hooks.

### PR DESCRIPTION
**Pull request by adam-vessey**
Ontime #1733.
https://jira.duraspace.org/browse/ISLANDORA-885

### CODE REVIEW

**Comment by adam-vessey** [comment]
Due to test using the islandora_object_access() and islandora_datastream_access() from https://github.com/Islandora/islandora/pull/352, the added test will be invalid until https://github.com/Islandora/islandora/pull/352 is merged.

File: api/tests/hooked_access.test [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+  }
+
+  /**
+   * Test all XACML operations on object.
+   */
+  public function testAllXacmlAccess() {
+    $user_allow = $this->user_allow;
+    $user_deny = $this->user_deny;
+
+    variable_set('islandora_strict_user_access_enforcement', FALSE);
```
Comment (jonathangreen) [inline api/tests/hooked_access.test]: Need to change this around. Since the variable doesn't exist anymore.

File: tests/hooked_access.test [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        // Meh... Either it didn't exist or the purge failed.
+      }
+    }
+  }
+
+  /**
+   * Test all operations on object.
+   */
+  public function testAllAccess() {
+    variable_set('islandora_strict_user_access_enforcement', FALSE);
```
Comment (jonathangreen) [inline tests/hooked_access.test]: and here.

**Comment by adam-vessey** [comment]
Removed the references to the strict check.

**Pull request merged**